### DISCUSSION TITLE
GitHub Sub-Directories

### DISCUSSION THREAD

**Discussion by svenbieg** [community/community#31409] [Other Feature Feedback, Questions, & Ideas]
I'm missing the feature that i have with Microsoft Team Foundation Server Version Control, to have access to sub-directories.
I like to clone sub-directories. I can download single files on GitHub, but there is no option to download a sub-directory.
I can't clone a sub-directory of a repository.
With Team Foundation Server this is possible.

Best regards,

Sven Bieg

**Comment by airtower-luna** [comment]
Technically, you can use [sparse checkout](https://www.git-scm.com/docs/git-sparse-checkout). However note that
* it's marked as experimental :warning: and
* Git will still need fetch much of the data.

My personal opinion is that if you need to check out part of the tree it's likely time to reconsider the repository structure, in particular if something should be in a separate repository.

**Reply by svenbieg** [reply]
With TFS i don't have to create a repository for each source-tree.
The sparse-checkout is much to complicated for me.
I think there should be a download-button in every sub-directory on GitHub.
